src/core/reference/include/ngraph/runtime/reference/grid_sample.hpp

```diff
@@ -109,8 +109,8 @@ DATA_ET reflection_data_with_align(const DATA_ET* data,
                                    long x_d) {
     const auto H = static_cast<long>(data_shape[2]);
     const auto W = static_cast<long>(data_shape[3]);
-    const auto H_2_2 = 2 * (H - 1);
-    const auto W_2_2 = 2 * (W - 1);
+    const auto H_2_2 = H == 1 ? 1 : 2 * (H - 1);
```

Comment:
Is this the same behavior as is in ONNX ?

Reply:
Yes. This value repeats the "pixel", what's the case here. Torch does the same.